### PR TITLE
#115 Updated React Configuration in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by cheesequake**
Updated CONTRIBUTING.md with the latest means of running the React frontend locally.

### CODE REVIEW

**Comment by AmanNegi** [comment]
Hey there, I was expecting to add different guides for different languages in the `CONTRIBUTING.md` file. So basically, the `CONTRIBUTING.md` file will contain **links** to, the React Setup Guide, NodeJS Setup Guide, Flutter Setup Guide, and other basic Styling, Code of Conduct, etc.

So could you update the Guide mentioned in https://github.com/AmanNegi/FreshNest/tree/main/frontend/react#readme and just link it to the `CONTRIBUTING.md`.

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -16,16 +16,18 @@
 
 1. Install Node.js and `npm` on your development machine by following the instructions on the Node.js website.
 2. Clone the project repository from GitHub.
-3. Open the project in your preferred IDE or code editor.
+3. Open the `frontend` folder in your preferred IDE or code editor.
```
Comment (AmanNegi) [inline docs/CONTRIBUTING.md:19]: Note, that we have to navigate to `fronted/react`.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -16,16 +16,18 @@
 
 1. Install Node.js and `npm` on your development machine by following the instructions on the Node.js website.
```
Comment (AmanNegi) [inline docs/CONTRIBUTING.md:17]: We need to mention pre-requisitives beforehand, like we need Node `v18.14.0` and `npm`. To use a local backend server the user might even be required to have `MongoDB` installed. So a section, by the name **pre-requisitives** could mention it at the start of this guide.

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -16,16 +16,18 @@
...
 2. Clone the project repository from GitHub.
-3. Open the project in your preferred IDE or code editor.
+3. Open the `frontend` folder in your preferred IDE or code editor.
 4. Install the required dependencies by running the **`npm install`** command in your terminal.
-5. Make changes to the code as needed, ensuring that you follow the project's coding standards and best practices.
-6. Next we will create a `.env` file, which contains the following key-value pairs:
+5. Next we will create a `.env` file, which contains the following key-value pairs:
```
Comment (AmanNegi) [inline docs/CONTRIBUTING.md:21]: See [here](https://github.com/AmanNegi/FreshNest/tree/main/frontend/react#readme), how it mentions we can use remote as well as local URL.

**Review by AmanNegi — COMMENTED** [review]
Hey there, I have added a few comments to this, but ensure these comments are taken care of in the [React Guide](https://github.com/AmanNegi/FreshNest/tree/main/frontend/react#readme). In this [file](https://github.com/AmanNegi/FreshNest/blob/dev-react/docs/CONTRIBUTING.md), you can simply add a link to the React Guide.

File: backend/README.md [hunk 4]
```diff
@@ -1,8 +1,49 @@
 # Backend For FreshNest
-> Node.js
+Prerequisites: 
+- Node.js [v18.4.0](https://nodejs.org/dist/v18.4.0/)
```
Comment (AmanNegi) [inline backend/README.md:3]: - Node.js [v18.14.0](https://nodejs.org/dist/v18.14.0/)

File: backend/README.md [hunk 5]
```diff
@@ -1,8 +1,49 @@
 # Backend For FreshNest
-> Node.js
+Prerequisites:
```
Comment (AmanNegi) [inline backend/README.md:2]: I would suggest you make the Prerequisites a subheading an `h2`, we can then have a similar **Steps** heading before the steps start.

Eg:
## Prerequisites:
## Steps

File: backend/README.md [hunk 6]
```diff
@@ -1,8 +1,49 @@
...
+5. Add an `.env` file in your `/backend` folder and add the below line:
+
+   ```env
+   DATABASE_URL="mongodb://127.0.0.1:27017/fresh-nest" # your MongoDB address with port and collection name
+   ```
+   
+6. Start the development server by running the following command:
+
+```
+npm start
```
Comment (AmanNegi) [inline backend/README.md:39]: For local development, I would suggest to run `npm run dev` as we have `nodemon` in our `dev-dependencies`.

File: frontend/react/README.md [hunk 7]
```diff
@@ -1,6 +1,9 @@
 ## Getting Started
+Prerequisites:
```
Comment (AmanNegi) [inline frontend/react/README.md:2]: Similar to the above, having a `Prerequisites` and `Steps` subheading(h2) would be helpful.

File: frontend/react/README.md [hunk 8]
```diff
@@ -11,37 +14,46 @@ git clone https://github.com/[your-username]/freshnest.git
 
 Replace [your-username] with your GitHub username.<br>
 
-3. Install Node.js [v18.4.0](https://nodejs.org/dist/v18.4.0/).<br>
```
Comment (AmanNegi) [inline frontend/react/README.md:14]: - Node.js [v18.14.0](https://nodejs.org/dist/v18.14.0/)

**Review by AmanNegi — CHANGES_REQUESTED** [review]
These are the last minor suggestions. Great work so far 🙂

**Comment by cheesequake** [comment]
@AmanNegi I have made the changes you had mentioned. Thank you!

**Review by AmanNegi — APPROVED** [review]
LGTM!

**Comment by AmanNegi** [comment]
Great work @cheesequake! Thanks a lot for contributing, Feel free to come back for more issues. 😃

**Pull request merged**